Updated Dockerfile

TachyonRSA opened the pull request:
- Updated dockerfile to use latest stable Ubuntu release
- Updated dockerfile to use the latest tor release

Commit:
Updated Dockerfile
- Updated dockerfile to use latest stable Ubuntu release
- Updated dockerfile to use the latest tor release

vpetersson commented on Dockerfile:
```diff
@@ -1,12 +1,20 @@
+# Finally install Tor
+RUN apt update
+RUN apt install sudo -y
+RUN apt install tor -y
```
The best practice in Docker is to do this in one go to avoid wasted layers (which means bigger disk footprint). Hence it's much better to chain all these commands as in the previous section.

TachyonRSA replied:
I was unaware of this practice, will update accordingly.

TachyonRSA replied:
@vpetersson - Fixed in 5b29ad4c9a47063b63f794f529987642acdb6562

Commit:
Removed Wasted Layers
- Removed wasted layers when setting GPG
- Removed wasted layers when installing sudo and tor

vpetersson commented on Dockerfile:
```diff
@@ -1,12 +1,18 @@
+RUN gpg --keyserver keys.gnupg.net --recv A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89 && gpg --export A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89 > tor.gpg && apt-key add tor.gpg
+
+# Finally install Tor
+RUN apt update && apt install sudo tor -y
```
Don't forget about `&& apt-get clean` as that cleans up the cache.

TachyonRSA replied:
Of course, apologies. Fixed in 0f6206ea9404421ba7c51cd55d76abfbd25eb03b

Commit:
Added cleaning
- Added apt clean to the dockerfile

vpetersson commented:
Thanks @TachyonRSA.
